### ISSUE TITLE
Feature request: one-class classification

### ISSUE THREAD

**Issue by cropmad1548** [tensorflow/tensorflow#4082]
Hello everyone.
I searched on the tutorials, examples and documentation, and I could not find anything related to this.
I am not sure if Tensorflow support this feature or not, but if it does, this issue is just about adding examples or documentation about this.

I tried to adapt the tutorial [Deep MNIST for Experts](https://www.tensorflow.org/versions/r0.10/tutorials/mnist/pros/index.html) to detect just one class, let's say detect if an image contains or not a kitty.

This is the prediction part of my code:

```
y_conv=tf.nn.softmax(tf.matmul(h_fc1_drop, W_fc2) + b_fc2)
cross_entropy = tf.reduce_mean(-tf.reduce_sum(y_ * tf.log(y_conv), reduction_indices=[1]))
train_step = tf.train.AdamOptimizer(1e-4).minimize(cross_entropy)
correct_prediction = tf.equal(tf.argmax(y_conv,1), tf.argmax(y_,1))
accuracy = tf.reduce_mean(tf.cast(correct_prediction, tf.float32))
sess.run(tf.initialize_all_variables())
for i in range(20000):
  batch = mnist.train.next_batch(50)
  if i%100 == 0:
    train_accuracy = accuracy.eval(feed_dict={
        x:batch[0], y_: batch[1], keep_prob: 1.0})
    print("step %d, training accuracy %g"%(i, train_accuracy))
  train_step.run(feed_dict={x: batch[0], y_: batch[1], keep_prob: 0.5})

print("test accuracy %g"%accuracy.eval(feed_dict={
    x: mnist.test.images, y_: mnist.test.labels, keep_prob: 1.0}))
```

The problem is that with one class, the softmax always return that class with confidence of 1, even for a blank image. I tried modifying the softmax and cross entropy, but I was not able to solve it.

I want to know what approach is recommended for this problem. I want the prediction to be the probability of an image being a kitty.

I know that this can be solved using a second label trained with random images, but I want to know if there is a better solution.

Thank you very much.
